liberty-maven-plugin/src/main/java/net/wasdev/wlp/maven/plugins/ApplicationXmlDocument.java

```diff
@@ -37,46 +37,55 @@ public ApplicationXmlDocument() {
             e.printStackTrace();
         }
     }
-    
-    public void createApplicationElement(String appFileName, String artifactId) {
+
+    public void createApplicationElement(String appFileName, String artifactId, boolean thin) {
         File app = new File(appFileName);
-        
-        switch (appFileName.substring(appFileName.lastIndexOf(".")+1)) {
-            case "war":
+
+        switch (appFileName.substring(appFileName.lastIndexOf(".") + 1)) {
+        case "war":
+            if (thin) {
+                createElement("springBootApplication", app, artifactId);
+            } else {
                 createElement("webApplication", app, artifactId);
-                break;
-            case "ear":
-                createElement("enterpriseApplication", app, artifactId);
-                break;
-            case "rar":
-                createElement("resourceAdapter", app, artifactId);
-                break;
-            default:
-                break;
+            }
+            break;
+        case "ear":
+            createElement("enterpriseApplication", app, artifactId);
+            break;
+        case "rar":
+            createElement("resourceAdapter", app, artifactId);
+            break;
+        case "jar":
+            if (thin) {
+                createElement("springBootApplication", app, artifactId);
+            }
+            break;
+        default:
```

Comment:
Just wondering if it is possible to get into this method with `thin` equal true for an ear, in which case maybe we should issue a warning that they configured something wrong and that their app will not be treated as a springBootApplication?

Reply:
It won't reach this stage. Since isSupportedType() throws an exception for the packaging types which does not support thin-project configuration